qiita_db/base.py

```diff
@@ -166,13 +166,30 @@ def __init__(self, id_):
 
         Parameters
         ----------
-        id_: the object identifier
+        id_: int, long, str, or unicode
+            the object identifier
 
         Raises
         ------
         QiitaDBUnknownIDError
             If `id_` does not correspond to any object
         """
+        # Most IDs in the database are numerical, but some (e.g., IDs used for
+        # the User object) are strings. Moreover, some integer IDs are passed
+        # as strings (e.g., '5'). Therefore, explicit type-checking is needed
+        # here to accommodate these possibilities.
+        id_type = type(id_)
+        if not isinstance(id_, (int, long, str, unicode)):
+            raise TypeError("id_ must be a numerical or text type (not %s) "
+                            "when instantiating %s" % (str(id_type),
+                                                       str(self.__class__)))
+
+        if id_type in (str, unicode):
```

Comment:
Yeah, probably more consistent that way. The behavior is slightly different (since `isinstance` returns true if it is one of the specified classes or any subclass), but I don't think it matters here.